File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        lst=list(s)
        vwl=['a','e','i','o','u','A','E','I','O','U']
        i=0
        j=len(s)-1
        while i<j:
            if(lst[i] in vwl):
                if lst[j] in vwl:
                    temp=lst[i]
                    lst[i]=lst[j]
                    lst[j]=temp
                    i=i+1
                    j=j-1
                else:
                    j=j-1
            elif lst[j] in vwl:
                i=i+1
            else:
                i=i+1
                j=j-1
        s=""
        for i in lst:
            s=s+i
        return s

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.cpp
class Solution {
public:
    string reverseVowels(string s) {
        int n=s.size();
        int c=0;
        string vow="aeiouAEIOU";
        string vec="";
        for(int i=n-1;i>=0;i--){
            if(vow.find(s[i])!=string::npos){
                vec+=(s[i]);
            }
        }
        int j=0;
        for(int i=0;i<n;i++){
            if(vow.find(s[i])!=string::npos){
                s[i]=vec[j++];
            }
        }
        return s;


    }
};